libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs

```diff
@@ -32,18 +32,15 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                 string readerValue = reader.Value.ToString();
                 return new ActivityTemplate((string)readerValue);
             }
-            else
+
+            JObject obj = JObject.Load(reader);
+            string kind = (string)obj["$kind"]; 
+            if (kind == "Microsoft.ActivityTemplate")
             {
-                JObject obj = JObject.Load(reader);
-                string kind = (string)obj["$kind"]; 
-                if (kind == "Microsoft.ActivityTemplate")
-                {
-                    return obj.ToObject<ActivityTemplate>();
-                }
-
-                var activity = obj.ToObject<Activity>();
-                return new StaticActivityTemplate((Activity)activity);
+                return obj.ToObject<ActivityTemplate>();
             }
+
+            return obj.ToObject<StaticActivityTemplate>();
```

Comment:
> return obj.ToObject<StaticActivityTemplate>(); [](start = 11, length = 47)

This is wrong. The serialized object is not a staticActivityTemplate, it's a StaticActivityTemplate wrapped around a serialized Activity.  I'm super suprised unit tests worked with this change.

Reply:
I had to do this to get tests\Microsoft.Bot.Builder.TestBot.Json\Samples\80 - AdaptiveSkillDialog\AdaptiveSkillDialog.main.dialog in TestJson to work, it was working until a while ago but not the project wasn't able to get the Activity property in AdaptiveSkillDialog.

Reply:
Reverted to your change from master that addresses the issue I was having loading the activity template.